src/hooks/loadables/useLoadBalances.ts

```diff
@@ -2,21 +2,22 @@ import { useEffect } from 'react'
 import { getBalances, type SafeBalanceResponse } from '@gnosis.pm/safe-react-gateway-sdk'
 import { useAppSelector } from '@/store'
 import useAsync, { type AsyncResult } from '../useAsync'
-import useSafeInfo from '../useSafeInfo'
 import { Errors, logError } from '@/services/exceptions'
 import { selectCurrency } from '@/store/settingsSlice'
+import { selectSafeInfo } from '@/store/safeInfoSlice'
 
 export const useLoadBalances = (): AsyncResult<SafeBalanceResponse> => {
-  const { safe, safeLoaded } = useSafeInfo()
+  // use the selector directly because useSafeInfo is memoized
+  const { data: safe } = useAppSelector(selectSafeInfo)
   const currency = useAppSelector(selectCurrency)
 
   // Re-fetch assets when the entire SafeInfo updates
-  const [data, error, loading] = useAsync<SafeBalanceResponse | undefined>(
-    async () => {
-      if (!safeLoaded) return
+  const [data, error, loading] = useAsync<SafeBalanceResponse>(
+    () => {
+      if (!safe) return
```

Comment:
Do we still need `safeLoaded` at all then?

Reply:
`safeLoaded` is needed in that hook, because there, `safe` is always defined but is a dummy object. It was done to avoid the `?` all the time.